### ISSUE TITLE
🎉 hot loading for markdown

### ISSUE THREAD

**Issue by MoOx** [MoOx/statinamic#11]
- use js instead of json
- use md to react instead of raw html https://www.npmjs.com/package/commonmark-react-renderer

**Comment by thangngoc89** [comment]
Any progress ?

**Comment by MoOx** [comment]
I will handle this after #23, #92 and #92. Then I will cut a release.
If you have idea(s), let's discuss about it :)

**Comment by thangngoc89** [comment]
Well. Maybe we should let webpack handle those file loading for us.  statinamic only need to keep track of a mapping between filename and url

This is a proofs of concept : 
webpack will generate N + 1 chunk (1 chunk per markdown files)

```
export default function loadAsync(filename, callback) {
  require([ "bundle!../content/" + filename + ".md" ], bundledResult => {
    bundledResult(result => callback(result))
  })
}
```

**Comment by MoOx** [comment]
It won't be that easy. Currently data are loaded via from json via a fetch() request. We need to rething a huge part. See https://github.com/MoOx/statinamic/pull/107#issuecomment-175195475

**Comment by thangngoc89** [comment]
Gatsby uses json https://github.com/gatsbyjs/gatsby/blob/2b0fffa0669d47f2798478f5dd22cd511cc83187/lib/loaders/markdown-loader/index.js#L38 and dangerouslySetInnerHTML https://github.com/gatsbyjs/gatsby/blob/2b0fffa0669d47f2798478f5dd22cd511cc83187/lib/isomorphic/wrappers/md.jsx#L17 but still get hot loading

**Comment by MoOx** [comment]
gatsby includes ALL content in one entry, this way, it's "easy".

**Comment by thangngoc89** [comment]
@MoOx yeah. That's why I deleted my comment.

**Comment by MoOx** [comment]
Haha ok :)

**Comment by thangngoc89** [comment]
My approach in #107 was using webpack code spliting (one markdown file per chunk)

But when it came to hot reload, I'm stuck

**Comment by MoOx** [comment]
The problem is that one route (PageContainer) require all md. I will think about that when #23 is fixed.

**Comment by thangngoc89** [comment]
Just wait for #23 to be fixed. I think I'll close #109. I mind have a better idea

**Comment by thangngoc89** [comment]
With #151 ,  now we have a map between real url and source file. I think I'll give this another try

**Comment by MoOx** [comment]
This should be pretty easy to handle in PageContainer using module.hot (just retrigger getPage).

**Comment by thangngoc89** [comment]
But it may require a little refactor, we need to require . md file somewhere.

**Comment by MoOx** [comment]
No. We can easily just retrigger the getPage in PageContainer, this will refetch the js every single HMR change (no big deal during development).
The hardest part will be more #67.

**Comment by thangngoc89** [comment]
Sound like a plan :) 

We can solve that temporary with html-to-react. But we need to solve #115 first

**Comment by MoOx** [comment]
Not sure.
I will try to add a stupid (but working) hot loading before cutting a new release.

**Comment by thangngoc89** [comment]
:+1:

**Comment by thangngoc89** [comment]
I wonder how can you do that ?

module.hot.accept require a valid `path`

```js
if (module.hot) {
  module.hot.accept('./some-path', newdata =>
    // do something
  )
}
```

**Comment by MoOx** [comment]
Closed by 8d72b30
